FTB/AssertionHelper.py

```diff
@@ -297,9 +297,10 @@ def escapePattern(msg):
 
     escapedStr = msg
 
-    activeChars = ["\\", "[", "]", "{", "}", "(", ")", "*", "+", "?", "^", "$", ".", "|"]
+    activeChars = ("\\", "[", "]", "{", "}", "(", ")", "*", "+", "?", "^", "$", ".", "|")
 
     for activeChar in activeChars:
-        escapedStr = escapedStr.replace(activeChar, "\\" + activeChar)
+        if activeChar in escapedStr:
```

Comment:
Is this really faster? replace is not doing a regex match and adding the additional if scans each string twice essentially instead of doing it once. I would assume that not changing the string at all is optimized and does not actually clone the string when assigning?

Reply:
Yep it is faster with one exception Python versions <2.7.15 in which case it is only faster on strings that are less than 200 characters. On newer Python versions I've seen speed ups of up to 6x.
So I guess we need to take in to account the Python version we run and the average string length that this function is used on? I put a bench-marking script on the server if you want to sanity check it.

At the very least a new string is created each iteration by `"\\" + activeChar`